### PR TITLE
fixes #10

### LINKED ISSUE #10: rewrite README.md
Current README.md is outdated. It shows initialization of `MyGower` which is a relict from the past.
It also gives too much theoretical introduction should be described elsewhere.

### PR DESCRIPTION
**Pull request by BartekKruczek**
- Metric description moved to separate section within documentation 
- KNN, hierarchical clustering and HDBSCAN moved to numerical comparisons within documentation 
- Simplified README, fix minor directory issues

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,16 +36,14 @@ Implementation of Gower's Metric in Python.
 
 Legend: ✅ - supported and tested, 🚧 - work in progress
 
-> [!Note]
-> The package is compatible with Python 3.10 to Python 3.13. Support for Python 3.14 is currently under development and will be available soon.
-> Any errors that occur the library might be associated with the incompatibility of the Python version.
-
 ## Documentation
 
 @TODO: add ref
```
Comment (mwardynski) [inline README.md]: don't we want to add the ref already?

File: README.md [hunk 2]
```diff
@@ -87,262 +90,10 @@ distance = gower(data[0], data[1])
 > [!Tip]
 > To calculate the pairwise distances for the entire dataset, you can do it manually or use an auxiliary function, like: [scipy.spatial.distance.pdist](https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.distance.pdist.html) or [sklearn.metrics.pairwise_distances](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.pairwise_distances.html).
 
-## Gower characteristics
```
Comment (mwardynski) [inline README.md:90]: We should mention `transform` in the basic example as well, don't you think?
Maybe with a comment, that it isn't necessary.

File: README.md (hunk 1 above)
Comment (mwardynski) [reply README.md]: you can now replace the placeholder with:
`https://gower-metric.readthedocs.io/en/latest/`

Comment (BartekKruczek) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (BartekKruczek) [reply README.md:90]: Done

**Review by mwardynski — APPROVED** [review]
